src/main/java/com/flightstats/hub/dao/aws/DynamoWebhookDaoLifecycle.java

```diff
@@ -0,0 +1,46 @@
+package com.flightstats.hub.dao.aws;
+
+import com.flightstats.hub.config.AppProperties;
+import com.flightstats.hub.config.DynamoProperties;
+import com.google.common.util.concurrent.AbstractIdleService;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+
+@Slf4j
+public class DynamoWebhookDaoLifecycle extends AbstractIdleService {
+    private final DynamoUtils dynamoUtils;
+    private final AppProperties appProperties;
+    private final DynamoProperties dynamoProperties;
+
+    @Inject
+    public DynamoWebhookDaoLifecycle(DynamoUtils dynamoUtils, AppProperties appProperties, DynamoProperties dynamoProperties) {
+        this.dynamoUtils = dynamoUtils;
+        this.appProperties = appProperties;
+        this.dynamoProperties = dynamoProperties;
+    }
+
+    @Override
+    protected void startUp() {
+        initialize();
+    }
+
+    @Override
+    protected void shutDown() {
+    }
+
+    private void initialize() {
+        if (appProperties.isReadOnly()) {
+            if (!dynamoUtils.doesTableExist(dynamoProperties.getWebhookConfigTableName())) {
+                String msg = String.format(
+                        "Probably fatal error. Dynamo webhook config table doesn't exist for r/o node.  %s",
+                        dynamoProperties.getWebhookConfigTableName());
+                log.error(msg);
+                throw new IllegalArgumentException(msg);
+            }
+            return;
```

Comment:
Personal style, but I'd use an else and remove this return to make it explicit that the two things are mutually exclusive situations. It can also avoid subtle bugs. 
Buggy code:
```
if (thing_is_true) {
   do thing;
   // oops...they coder forgot to put a return here and it ends up always executing the statement below!
}
do_mutually_exclusive_thing;
```
Avoids even the possibility of that bug:
```
if (thing_is_true) {
    do thing;
}  else {
    do mutually exclusive thing;
}
```

Reply:
Yep, done.